mobile_verifier/src/boosting_oracles/mod.rs

```diff
@@ -1,67 +1,388 @@
+            tracing::info!("Found initial {} data set: {}", T::TYPE.to_prefix(), data_set_path.to_string_lossy());
+            self.data_set.lock().await.update(&data_set_path, time_to_use)?;
+        }
+
+        // Another option I considered instead of polling was to use ENOTIFY, but that seemed
+        // not as good, as it would hog a pool connection.
+        //
+        // We set the poll duration to 30 minutes since that seems fine.
+        let poll_duration = Duration::minutes(30);
+
+        loop {
+            // Find the latest urbanization file
+            tracing::info!("Checking for new data sets");
+            let mut data_set = self.data_set.lock().await;
+            let curr_data_set = data_set.timestamp();
+            let latest_data_set: Option<NewDataSet> = sqlx::query_as(
+                "SELECT filename, time_to_use, status FROM data_sets WHERE status != 'processed' AND data_set = $1 AND COALESCE(time_to_use > $2, TRUE) AND time_to_use <= $3 ORDER BY time_to_use DESC LIMIT 1"
+            )
+                .bind(T::TYPE)
+                .bind(curr_data_set)
+                .bind(Utc::now())
+                .fetch_optional(&self.pool)
+                .await?;
+
+            if let Some(latest_data_set) = latest_data_set {
+                let path = self.get_data_set_path(latest_data_set.time_to_use);
+
+                // Download the file if it hasn't been downloaded already:
+                if !latest_data_set.status.is_downloaded() {
+                    tracing::info!("Downloading new data set: {}", path.to_string_lossy());
+                    // TODO: abstract this out to a function
+                    let stream = self.store.get_raw(latest_data_set.filename.clone()).await?;
+                    let mut bytes = tokio_util::codec::FramedRead::new(
+                        async_compression::tokio::bufread::GzipDecoder::new(
+                            tokio_util::io::StreamReader::new(stream),
+                        ),
+                        tokio_util::codec::BytesCodec::new(),
+                    );
+                    let mut file = File::create(&path).await?;
+                    while let Some(bytes) = bytes.next().await.transpose()? {
+                        file.write_all(&bytes).await?;
+                    }
+                    // Set the status to be downloaded
+                    sqlx::query("UPDATE data_sets SET status = 'downloaded' WHERE filename = $1")
+                        .bind(&latest_data_set.filename)
+                        .execute(&self.pool)
+                        .await?;
+                }
+
+                // Now that we've downloaded the file, load it into the data set
+                data_set.update(Path::new(&path), latest_data_set.time_to_use)?;
+
+                drop(data_set);
+
+                // Update the hexes
+                let boosting_reports = set_oracle_boosting_assignments(
+                    UnassignedHex::fetch_all(&self.pool),
```

Comment:
Does this mean that every time the mobile-verifier boots it will update all the hexes with the latest dataset assignments?

Reply:
after re-reading I see that it doesn't since if won't find a non processed dataset in the db.  I think we really need to work on making this more readable.